### PR TITLE
Mnash 5332 update default subscription

### PR DESCRIPTION
**Pull request by m-nash**

### CODE REVIEW

File: Azure.ResourceManager.Core/AzureResourceManagerClient.cs [hunk 1]
```diff
@@ -179,43 +166,12 @@ public T GetResourceOperations<T>(string subscription, string resourceGroup, str
...
-        /// <summary>
-        /// Gets default subscription.
-        /// </summary>
-        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="P:System.Threading.CancellationToken.None" />. </param>
-        /// <returns> A <see cref="Task"/> that on completion returns the subscription id. </returns>
-        internal async Task<Subscription> GetDefaultSubscriptionAsync(CancellationToken cancellationToken = default)
+        private Subscription GetDefaultSubscription()
         {
-            return await GetSubscriptionContainer().GetDefaultSubscriptionAsync(cancellationToken);
+            var sub = GetSubscriptionContainer().List().FirstOrDefault();
```
Comment (bquantump) [inline Azure.ResourceManager.Core/AzureResourceManagerClient.cs:171]: So is default subscription the first element of the result?

File: Azure.ResourceManager.Core/AzureResourceManagerClient.cs [hunk 2]
```diff
@@ -179,43 +166,12 @@ public T GetResourceOperations<T>(string subscription, string resourceGroup, str
...
+            if (sub is null)
```
Comment (bquantump) [inline Azure.ResourceManager.Core/AzureResourceManagerClient.cs:172]: using "is" to avoid using "==" due to overload?

**Review by bquantump — APPROVED** [review]

File: Azure.ResourceManager.Core/AzureResourceManagerClient.cs (hunk 1 above)
Comment (m-nash) [reply Azure.ResourceManager.Core/AzureResourceManagerClient.cs:171]: yes it returns in a repeatable pattern

File: Azure.ResourceManager.Core/AzureResourceManagerClient.cs (hunk 2 above)
Comment (m-nash) [reply Azure.ResourceManager.Core/AzureResourceManagerClient.cs:172]: yes

**Pull request merged**